service/controller/clusterapi/v29/resource/tccp/template_render_test.go

```diff
@@ -0,0 +1,273 @@
+			Region: "eu-central-1",
+		},
+	}
+
+	cr := v1alpha1.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Labels: map[string]string{
+				label.Cluster: TestClusterID,
+			},
+		},
+	}
+
+	return withG8sClusterSpec(cr, g8sSpec)
+}
+
+func defaultControllerContext() controllercontext.Context {
+	return controllercontext.Context{
+		Status: controllercontext.ContextStatus{
+			ControlPlane: controllercontext.ContextStatusControlPlane{
+				AWSAccountID: "control-plane-account",
+				NATGateway:   controllercontext.ContextStatusControlPlaneNATGateway{},
+				RouteTable:   controllercontext.ContextStatusControlPlaneRouteTable{},
+				PeerRole: controllercontext.ContextStatusControlPlanePeerRole{
+					ARN: "imaginary-cp-peer-role-arn",
+				},
+				VPC: controllercontext.ContextStatusControlPlaneVPC{
+					CIDR: "10.1.0.0/16",
+				},
+			},
+			TenantCluster: controllercontext.ContextStatusTenantCluster{
+				AvailabilityZones:     []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+				AWSAccountID:          "tenant-account",
+				Encryption:            controllercontext.ContextStatusTenantClusterEncryption{},
+				HostedZoneNameServers: "1.1.1.1,8.8.8.8",
+				MasterInstance:        controllercontext.ContextStatusTenantClusterMasterInstance{},
+				TCCP: controllercontext.ContextStatusTenantClusterTCCP{
+					ASG:               controllercontext.ContextStatusTenantClusterTCCPASG{},
+					IsTransitioning:   false,
+					MachineDeployment: defaultMachineDeployment(),
+					VPC: controllercontext.ContextStatusTenantClusterTCCPVPC{
+						PeeringConnectionID: "imagenary-peering-connection-id",
+					},
+				},
+				VersionBundleVersion: "6.3.0",
+				WorkerInstance: controllercontext.ContextStatusTenantClusterWorkerInstance{
+					DockerVolumeSizeGB: "100",
+					CloudConfigVersion: key.CloudConfigVersion,
+					Image:              "ami-0eb0d9bb7ad1bd1e9",
+					Type:               "m5.xlarge",
+				},
+			},
+		},
+	}
+}
+
+func defaultMachineDeployment() v1alpha1.MachineDeployment {
+	g8sSpec := g8sv1alpha1.AWSMachineDeploymentSpec{
+		NodePool: g8sv1alpha1.AWSMachineDeploymentSpecNodePool{
+			Description: "Test node pool for cluster in template rendering unit test.",
+			Machine: g8sv1alpha1.AWSMachineDeploymentSpecNodePoolMachine{
+				DockerVolumeSizeGB:  100,
+				KubeletVolumeSizeGB: 100,
+			},
+			Scaling: g8sv1alpha1.AWSMachineDeploymentSpecNodePoolScaling{
+				Max: 5,
+				Min: 3,
+			},
+		},
+		Provider: g8sv1alpha1.AWSMachineDeploymentSpecProvider{
+			AvailabilityZones: []string{
+				"eu-central-1a",
+				"eu-central-1c",
+			},
+			Worker: g8sv1alpha1.AWSMachineDeploymentSpecProviderWorker{
+				InstanceType: "m5.2xlarge",
+			},
+		},
+	}
+
+	cr := v1alpha1.MachineDeployment{
+		ObjectMeta: v1.ObjectMeta{
+			Annotations: map[string]string{
+				annotation.MachineDeploymentSubnet: "10.100.8.0/24",
+			},
+			Labels: map[string]string{
+				label.Cluster: TestClusterID,
+				//label.MachineDeployment: TestMachineDeploymentID,
```

Comment:
I assume this label is missing because of some unmerged PR?

Reply:
Yeah, that label doesn't exist yet, but it will be there so I just added it in comments in this case.

Reply:
https://github.com/giantswarm/aws-operator/blob/master/pkg/label/label.go#L9

Reply:
Started in different branch and then made PR against master. Oh well. Fixed. :)